### PR TITLE
Adding analytics logging to new school association flow

### PR DESCRIPTION
**Pull request by hannahbergam**
This PR instruments events on the new school association flow (which will not start sending until the AB test starts). 

There is a commented out event that will eventually represent the dropdown being 'opened', but that is a very tricky engineering task to implement. More conversation in the product spec [here](https://docs.google.com/document/d/1Ulaxj3K6AMNOOjw_244LNOV5ypuHJMjGpUItXpZuC4I/edit?disco=AAABJx8KQyg).  

I made a small abstraction to keep the school function readible and reduce each analytics call to one line (before, each was 5 lines). 

The rest of the events should be self explanatory and I was able to tour them, as evidenced here:
Screenshot 2024-04-05 at 12 59 56 PM

## Warning!!

The [AP CSP Create Performance Task](https://apcentral.collegeboard.org/courses/ap-computer-science-principles/exam) is in progress. The most critical dates are from April 3 - April 30, 2024. Please consider any risk introduced by this PR that could affect our students taking AP CSP. Code.org students taking AP CSP primarily use App Lab for their Create Task, however a small percent use Game Lab. Carefully consider whether your change has any risk of alterering, changing, or breaking anything in these two labs. Even small changes, such as a different button color, are considered significant during this time period. Reach out to the Student Learning team or Curriculum team for more details.

## Links

- spec: https://docs.google.com/document/d/1Ulaxj3K6AMNOOjw_244LNOV5ypuHJMjGpUItXpZuC4I/edit
- jira ticket: https://codedotorg.atlassian.net/browse/ACQ-1607
- https://codedotorg.atlassian.net/browse/ACQ-1719

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/SchoolZipSearch.jsx [hunk 1]
```diff
@@ -35,12 +39,29 @@ export default function SchoolZipSearch({fieldNames, zip}) {
...
     const schoolId = e.target.value;
-    setSelectedSchoolNcesId(schoolId);
-    if (schoolId === 'clickToAdd') {
+    let ncesId;
+    if (schoolId === NO_SCHOOL_SETTING) {
+      ncesId = '';
+      sendAnalyticsEvent(EVENTS.DO_NOT_TEACH_AT_SCHOOL_CLICKED, {});
+    } else if (schoolId === SELECT_A_SCHOOL) {
+      ncesId = '';
+      // sendAnalyticsEvent(EVENTS.SCHOOL_LIST_OPENED, {});
```
Comment (bethanyaconnor) [inline apps/src/templates/SchoolZipSearch.jsx:54]: should this be commented out?

**Review by bethanyaconnor — APPROVED** [review]
Looks good but not sure if one line should be commented out?

File: apps/src/templates/SchoolZipSearch.jsx (hunk 1 above)
Comment (hannahbergam) [reply apps/src/templates/SchoolZipSearch.jsx:54]: Yes- there's a nod to it in the PR description.. I'm waiting to hear back from Dani about how this should go so I have a placeholder here for now (to get the other events sending).

Comment (bethanyaconnor) [reply apps/src/templates/SchoolZipSearch.jsx:54]: 🤦‍♀️ sorry, reading too fast trying to clear out reviews before the end of my day. Makes sense.

Comment (hannahbergam) [reply apps/src/templates/SchoolZipSearch.jsx:54]: All good!!

**Pull request merged**